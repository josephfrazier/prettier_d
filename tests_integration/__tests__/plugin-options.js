"use strict";

const runPrettier = require("../runPrettier");
const snapshotDiff = require("snapshot-diff");

describe("show external options with `--help`", () => {
  const originalStdout = runPrettier("plugins/options", ["--help"]).stdout;
  const pluggedStdout = runPrettier("plugins/options", [
    "--help",
    "--plugin=./plugin"
  ]).stdout;
  expect(snapshotDiff(originalStdout, pluggedStdout)).toMatchSnapshot();
});

describe("show detailed external option with `--help foo-option`", () => {
  runPrettier("plugins/options", [
    "--plugin=./plugin",
    "--help",
    "foo-option"
  ]).test({
    status: 0
  });
});

describe("external options from CLI should work", () => {
  runPrettier(
    "plugins/options",
    [
      "--plugin=./plugin",
      "--stdin-filepath",
      "example.foo",
      "--foo-option",
      "baz"
    ],
    { input: "hello-world" }
  ).test({
    stdout: "foo:baz",
    stderr: "",
    status: 0,
    write: []
  });
});

describe("external options from config file should work", () => {
  runPrettier(
    "plugins/options",
    ["--config=./config.json", "--stdin-filepath", "example.foo"],
    { input: "hello-world" }
  ).test({
    stdout: "foo:baz",
    stderr: "",
    status: 0,
    write: []
  });
});
