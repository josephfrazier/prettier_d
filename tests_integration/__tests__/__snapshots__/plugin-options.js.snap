// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -20,18 +20,20 @@
                             Control how Prettier formats quoted code embedded in the file.
                             Defaults to auto.
    --end-of-line <lf|crlf|cr|auto>
                             Which end of line characters to apply.
                             Defaults to lf.
+   --foo-option <bar|baz>   foo description
+                            Defaults to bar.
    --html-whitespace-sensitivity <css|strict|ignore>
                             How to handle whitespaces in HTML.
                             Defaults to css.
    --jsx-bracket-same-line  Put > on the last line instead of at a new line.
                             Defaults to false.
    --jsx-single-quote       Use single quotes in JSX.
                             Defaults to false.
-   --parser <flow|babel|babel-flow|babel-ts|typescript|css|less|scss|json|json5|json-stringify|graphql|markdown|mdx|vue|yaml|html|angular|lwc>
+   --parser <flow|babel|babel-flow|babel-ts|typescript|css|less|scss|json|json5|json-stringify|graphql|markdown|mdx|vue|yaml|html|angular|lwc|foo-parser>
                             Which parser to use.
    --print-width <int>      The line length where Prettier will try wrap.
                             Defaults to 80.
    --prose-wrap <always|never|preserve>
                             How to wrap prose."
`;

exports[`include plugin's parsers to the values of the \`parser\` option\` (stderr) 1`] = `""`;

exports[`include plugin's parsers to the values of the \`parser\` option\` (stdout) 1`] = `
"--parser <flow|babel|babel-flow|babel-ts|typescript|css|less|scss|json|json5|json-stringify|graphql|markdown|mdx|vue|yaml|html|angular|lwc|foo-parser>

  Which parser to use.

Valid options:

  flow             Flow
  babel            JavaScript
  babel-flow       Flow
  babel-ts         TypeScript
  typescript       TypeScript
  css              CSS
  less             Less
  scss             SCSS
  json             JSON
  json5            JSON5
  json-stringify   JSON.stringify
  graphql          GraphQL
  markdown         Markdown
  mdx              MDX
  vue              Vue
  yaml             YAML
  html             HTML
  angular          Angular
  lwc              Lightning Web Components
  foo-parser       foo (plugin: ./plugin)
"
`;

exports[`include plugin's parsers to the values of the \`parser\` option\` (write) 1`] = `Array []`;

exports[`show detailed external option with \`--help foo-option\` (stderr) 1`] = `""`;

exports[`show detailed external option with \`--help foo-option\` (stdout) 1`] = `
"--foo-option <bar|baz>

  foo description

Valid options:

  bar   bar description
  baz   baz description

Default: bar
"
`;

exports[`show detailed external option with \`--help foo-option\` (write) 1`] = `Array []`;
